main.nf

```diff
@@ -845,40 +1199,63 @@ process madeline {
 	output:
 		file("${ped}.madeline.xml") into madeline_ped
 		set group, file("${group}_madde.INFO") into madde_INFO
+		set group, file("*versions.yml") into ch_madeline_versions
 
 	when:
 		mode == "family" && params.assay == "wgs"
 
+	script:
+		"""
+		source activate tools
+		ped_parser \\
+			-t ped $ped \\
+			--to_madeline \\
+			-o ${ped}.madeline
+		madeline2 \\
+			-L "IndividualId" ${ped}.madeline \\
+			-o ${ped}.madeline \\
+			-x xml
+		echo "MADDE	$type ${params.accessdir}/ped/${ped}.madeline.xml" > ${group}_madde.INFO
+
+		${madeline_version(task)}
+		"""
+
+	stub:
+		"""
+		source activate tools
+		touch ${group}_madde.INFO
+		touch ${ped}.madeline.xml
+
+		${madeline_version(task)}
+		"""
+}
+def madeline_version(task) {
 	"""
-	source activate tools
-	ped_parser \\
-		-t ped $ped \\
-		--to_madeline \\
-		-o ${ped}.madeline
-	madeline2 \\
-		-L "IndividualId" ${ped}.madeline \\
-		-o ${ped}.madeline \\
-		-x xml
-	echo "MADDE	$type ${params.accessdir}/ped/${ped}.madeline.xml" > ${group}_madde.INFO
+	cat <<-END_VERSIONS > ${task.process}_versions.yml
+	${task.process}:
+	    ped-parser: \$(echo \$(ped_parser --version 2>&1) | sed -e "s/^.*ped_parser version: //")
+	    madeline2: \$(echo \$(madeline --version 2>&1) | grep : | sed -e"s/^.*Madeline //; s/PDE : 1.*//")
```

Comment:
```suggestion
	    madeline2: \$(echo \$(madeline2 --version 2>&1) | grep : | sed -e"s/^.*Madeline //; s/PDE : 1.*//")
```

Wrong command.  Prior to this change the code above would silently fail and just print an error message for madeline in versions_all.yml instead of a version number.

Reply:
OK, good that you spotted it. This is my mistake. Looks like I updated the key to "madeleine2" rather than the command

Reply:
Updated